### DISCUSSION TITLE
Need working Letsencrypt docker-compose.yaml and traefik.yaml

### DISCUSSION THREAD

**Discussion by wicknet** [photoprism/photoprism#2465] [Q&A]
* Photoprism installed on Ubuntu 20.04 server in single click via https://dl.photoprism.app/docker/cloud/setup.sh
* Server's internal IP mapped to public IP with port 80/443 open for egress and ingress
* External DNS updated to map photos.mydomain.com to public IP of server
* All works well with built-in self signed certs, including access via internal IP or via FQDN from Internet

All attempts to follow the inline documentation as well as generic samples on Traefik site to generate a SSL Letsencrypt cert result in no errors on docker-compse up -d command BUT after that dashboard is inaccessible even via internal IP with "Bad gateway" error showing on loading web page.

Only inconsistency I see is that letsencrypt.json resides in /opt/traefik directory, but is being called from storage: /data/letsencrypt.json in traefik.yaml

Can the brain trust share a copy of their docker-compose.yaml and traefik.yaml that is working with Letsencrypt, after masking your domain name and passwords?

Thanks!

**Comment by r-erd** [comment]
I am currently facing the same issue and I'm curious if you solved it by now. Maybe you could share your files.

Thanks!

**Comment by wicknet** [comment]
Sure.. here's are my steps for a manual Letsencrypt setup for PhotoPrism

1. Enable port 80 for server on firewall (ufw allow from any to any port 80)

2. Stop docker first so port 80 becomes available and open it on firewall

3. Run the following commands to stop docker and install classic certbot via snap

**cd /opt/photoprism
docker-compose stop
cd /root
apt install snapd
sudo snap install core; sudo snap refresh core sudo snap install --classic certbot
sudo ln -s /snap/bin/certbot /usr/bin/certbot**

4. Use certbot command below to generate a new cert. When prompted, enter fqdn of server (eg. photos.yourdomain.com) and use port 80 to spin up a new server. Certs will be saved in /etc/letsencrypt/live/photos.yourdomain.com/

**certbot certonly**

5. Test automatic renewal with command below, after stopping docker-compose to free up port 80

**certbot renew --dry-run**

6. Copy over new or updated cert and key to Photoprism using commands below, overwriting its default cert/key (this step is crucial. DO NOT use your own cert name or key)

**cp /etc/letsencrypt/live/photos.yourdomain.com/fullchain.pem /opt/photoprism/certs/cert.crt
cp /etc/letsencrypt/live/photos.yourdomain.com/privkey.pem /opt/photoprism/certs/cert.key**

6. Start docker again using commands below

**cd /opt/photoprism
docker-compose  up -d**

7. Close Port 80 on firewall

8. Create a file in /root folder to auto renew certs using commands below:

_nano certrenew.sh_

**#!/bin/bash
cd /opt/photoprism
docker-compose stop
sleep 1m
cd /root
certbot renew
sleep 2m
cp /etc/letsencrypt/live/photos.yourdomain.com/fullchain.pem /opt/photoprism/certs/cert.crt
cp /etc/letsencrypt/live/photos.yourdomain.com/privkey.pem /opt/photoprism/certs/cert.key
cd /opt/photoprism
docker-compose  up -d**

9. Set up cron job to renew cert every 1st of month using commands below:

**crontab -e
0 0 1 * * /root/certrenew.sh >> /var/log/cron.log 2>&1**

10. Done. Reboot and enjoy a SSL secured server hosting your pictures.

**Reply by r-erd** [reply]
Thanks a lot. I assume you used the `traefik.yaml` as supplied [here](https://docs.photoprism.app/getting-started/proxies/traefik/) and also only added the changes that are suggested there to the `docker-compose.yaml`? I will try it later today.
